### PR TITLE
Windows

### PR DESCRIPTION
**Pull request by diorcety**
This is the fixes for better support of Windows
- Use the compiler used by numpy
- Support only OMP 2.0 (MSVC ...)
- Not so nice way to get the OMP library with MSVC (but working)

### CODE REVIEW

File: omp/__init__.py [hunk 1]
```diff
@@ -25,6 +26,26 @@ class OpenMP(object):
     """
 
     def __init__(self):
+        if c_compiler is None or c_compiler.compiler_type != 'msvc':
```
Comment (serge-sans-paille) [inline omp/__init__.py]: I don't like that. If ``c_compiler.compiler_type == 'msvc'``, what is the type : value of ``c_compiler`` ?

**Comment by diorcety** [comment]
Certainly instance MSVCCompiler

**Comment by diorcety** [comment]
But it seems the usage of compiler_type is  the "official" way

File: omp/__init__.py [hunk 2]
```diff
@@ -25,6 +26,26 @@ class OpenMP(object):
...
+            self.init_not_msvc()
+        else:
+            self.init_msvc()
+
+    def init_msvc(self):
+        vcomp_path = None
+        for i in range(15, 8, -1):
```
Comment (serge-sans-paille) [inline omp/__init__.py]: This scanning looks very unrliable in the future....

File: pythran/tables.py [hunk 3]
```diff
@@ -4424,6 +4414,27 @@ def partialsum(seq):
     del MODULES['operator_']['matmul']
     del MODULES['operator_']['__matmul__']
 
+# OMP version
```
Comment (serge-sans-paille) [inline pythran/tables.py]: why do you need to do that instead of the previous scanism mechanism?

Comment (diorcety) [reply pythran/tables.py]: ? I don't remember, but some module use the table for getting omp function then fail on Windows because the function doesn't existe on omp 2.0

File: omp/__init__.py (hunk 2 above)
Comment (diorcety) [reply omp/__init__.py]: Yes, I need to fond a better way

Comment (diorcety) [reply omp/__init__.py]: something like that: https://github.com/mhammond/pywin32/blob/master/setup.py#L1060 ?

Comment (serge-sans-paille) [reply omp/__init__.py]: That also sounds super hacky. I prefer your scanning after all, but maybe trying a given version number based on the compiler path first?

File: pythran/tables.py (hunk 3 above)
Comment (diorcety) [reply pythran/tables.py]: Yes msvc omp support is 2.0, not more

File: omp/__init__.py (hunk 2 above)
Comment (diorcety) [reply omp/__init__.py]: I can try to use compiler.__version or msvc_runtime_major from misc_util (numpy)?

**Comment by serge-sans-paille** [comment]
yep!

**Comment by diorcety** [comment]
Then, I will try to rebase and cleanup this PR soon.

File: pythran/pythonic/include/utils/numpy_conversion.hpp [hunk 4]
```diff
@@ -5,6 +5,14 @@
...
 #include <utility>
 
+#if _MSC_VER
+#define NUMPY_EXPR_TO_NDARRAY0_DECL(fname)                                     \
+  template <class E, class... Types,                                           \
+      typename std::enable_if<                                                 \
+          !types::is_ndarray<E>::value && types::is_array<E>::value,           \
+          E>::type* = nullptr>                                                 \
+  auto fname(E const &expr, Types &&... others);
+#else
```
Comment (serge-sans-paille) [inline pythran/pythonic/include/utils/numpy_conversion.hpp]: This seems unrelated to the other changes

File: pythran/toolchain.py [hunk 5]
```diff
@@ -314,17 +315,24 @@ def compile_cxxfile(module_name, cxxfile, output_binary=None, **kwargs):
...
+        if f.endswith(ext):
+            if not output_binary:
+                output_binary = os.path.join(os.getcwd(), module_name + ext)
+            copy(f, output_binary)
+        else:
+            if not output_binary:
+                output_directory = os.getcwd()
+            else:
+                output_directory = os.path.dirname(output_binary)
+            copy(f, os.path.join(output_directory, os.path.basename(f)))
```
Comment (serge-sans-paille) [inline pythran/toolchain.py]: This deserve at least a comment. Maybe hardcode the possible extension to copy instead of globbing? This would make the goal more explicit

File: pythran/pythonic/include/utils/numpy_conversion.hpp (hunk 4 above)
Comment (diorcety) [reply pythran/pythonic/include/utils/numpy_conversion.hpp]: This allow to compile with VS2015/VS2017. This is Windows related :)

File: pythran/toolchain.py (hunk 5 above)
Comment (diorcety) [reply pythran/toolchain.py]: Sure

File: pythran/backend.py [hunk 6]
```diff
@@ -914,7 +914,7 @@ def visit_Num(self, node):
 
     def visit_Str(self, node):
         quoted = node.s.replace('"', '\\"').replace('\n', '\\n"\n"')
-        return '"' + quoted + '"'
+        return 'pythonic::types::str("' + quoted + '")'
```
Comment (diorcety) [inline pythran/backend.py]: error C2666: '`anonymous-namespace'::pythonic::types::dict<`anonymous-namespace'::pythonic::types::str,`anonymous-namespace'::pythonic::types::str>::operator []': 5 overloads have similar conversions
D:\tmp\xalgoenv\lib\site-packages\pythran-0.8.6.dev0+x-py3.6.egg/pythran\pythonic\include\types\dict.hpp(178): note: could be 'const `anonymous-namespace'::pythonic::types::str &`anonymous-namespace'::pythonic::types::dict<`anonymous-namespace'::pythonic::types::str,`anonymous-namespace'::pythonic::types::str>::operator [](const K &) const'
        with
        [
            K=`anonymous-namespace'::pythonic::types::str
        ]
D:\tmp\xalgoenv\lib\site-packages\pythran-0.8.6.dev0+x-py3.6.egg/pythran\pythonic\include\types\dict.hpp(172): note: or       '`anonymous-namespace'::pythonic::types::str &`anonymous-namespace'::pythonic::types::dict<`anonymous-namespace'::pythonic::types::str,`anonymous-namespace'::pythonic::types::str>::operator [](const K &)'
        with
        [
            K=`anonymous-namespace'::pythonic::types::str
        ]
D:\tmp\xalgoenv\lib\site-packages\pythran-0.8.6.dev0+x-py3.6.egg/pythran\pythonic\include\types\dict.hpp(180): note: or       'const V &`anonymous-namespace'::pythonic::types::dict<`anonymous-namespace'::pythonic::types::str,V>::operator []<char[9]>(const OtherKey (&)) const'
        with
        [
            V=`anonymous-namespace'::pythonic::types::str,
            OtherKey=char [9]
        ]
D:\tmp\xalgoenv\lib\site-packages\pythran-0.8.6.dev0+x-py3.6.egg/pythran\pythonic\include\types\dict.hpp(174): note: or       'V &`anonymous-namespace'::pythonic::types::dict<`anonymous-namespace'::pythonic::types::str,V>::operator []<char[9]>(const OtherKey (&))'
        with
        [
            V=`anonymous-namespace'::pythonic::types::str,
            OtherKey=char [9]
        ]
xalgo/_comp.cpp(91): note: or       'built-in C++ operator[(__int64, char [9])'
xalgo/_comp.cpp(91): note: while trying to match the argument list '(`anonymous-namespace'::pythonic::types::dict<`anonymous-namespace'::pythonic::types::str,`anonymous-namespace'::pythonic::types::str>, char [9])'

File: pythran/pythonic/include/utils/numpy_conversion.hpp (hunk 4 above)
Comment (diorcety) [reply pythran/pythonic/include/utils/numpy_conversion.hpp]: Otherwise ... 
fatal error C1202: recursive type or function dependency context too complex

**Comment by diorcety** [comment]
Let me know if you want more information

**Comment by serge-sans-paille** [comment]
> Let me know if you want more information
Appveyor fails on ``ssize_t`` redefinition, can you fix that (and probably rebase on master too).

**Comment by serge-sans-paille** [comment]
Obsoleted by #1013
